sessions/payload.go

```diff
@@ -0,0 +1,77 @@
+package sessions
+
+import (
+	"os"
+	"runtime"
+	"time"
+)
+
+// notifierPayload defines the .notifier subobject of the payload
+type notifierPayload struct {
+	Name    string `json:"name"`
+	URL     string `json:"url"`
+	Version string `json:"version"`
+}
+
+// appPayload defines the .app subobject of the payload
+type appPayload struct {
+	Type         string `json:"type"`
+	ReleaseStage string `json:"releaseStage"`
+	Version      string `json:"version"`
+}
+
+// devicePayload defines the .device subobject of the payload
+type devicePayload struct {
+	OsName   string `json:"osName"`
+	Hostname string `json:"hostname"`
+}
+
+// sessionCountsPayload defines the .sessionCounts subobject of the payload
+type sessionCountsPayload struct {
+	StartedAt       string `json:"startedAt"`
+	SessionsStarted int    `json:"sessionsStarted"`
+}
+
+// sessionPayload defines the top level payload object
+type sessionPayload struct {
+	Notifier      notifierPayload        `json:"notifier"`
+	App           appPayload             `json:"app"`
+	Device        devicePayload          `json:"device"`
+	SessionCounts []sessionCountsPayload `json:"sessionCounts"`
+}
+
+// makeSessionPayload creates a sessionPayload based off of the given sessions and config
+func makeSessionPayload(sessions []*Session, config *SessionTrackingConfiguration) *sessionPayload {
+	releaseStage := config.ReleaseStage
+	if releaseStage == "" {
+		releaseStage = "production"
+	}
+	hostname := config.Hostname
+	if hostname == "" {
+		hostname, _ = os.Hostname() //Ignore the hostname if this call errors
```

Comment:
have we considered caching this value? See https://thoeni.io/post/macos-sierra-java/